### PR TITLE
Dprintf optimization

### PR DESCRIPTION
**Pull request by PeterSolMS**
Here's the initial implementation for putting the stress log into a memory mapped file and piping the dprintf calls to the stress log.

It's much faster than the dprintf implementation we normally and doesn't take a lock.

There is logic inside StressLog.cpp to switch to the memory mapped file when COMPLUS_StressLogFilename is set. Here's a set of environment variables I've used:

COMPlus_StressLogFilename=GCPerfSimStresslog.log
COMPlus_gcServer=1
COMPlus_LogLevel=9
COMPlus_StressLog=1
COMPlus_StressLogSize=10000000
COMPlus_TotalStressLogSize=1000000000

So this creates a max of 64 GB of stress log.

This mechanism also works for clrgc.dll, which accounts for some complications.

I have an analyzer to go with this that I'm still working on...

### CODE REVIEW

File: src/coreclr/gc/gc.cpp [hunk 1]
```diff
@@ -9283,8 +9283,14 @@ size_t gc_heap::sort_mark_list()
         return 0;
     }
 
+    uint64_t now = GetHighPrecisionTimeStamp();
```
Comment (Maoni0) [inline src/coreclr/gc/gc.cpp:9286]: since this is only used in dprintf you could put it under #ifdef TRACE_GC

**Review by Maoni0 — APPROVED** [review]
LGTM (the changes in gcpriv.h are still temporary right?)

File: src/coreclr/gc/gcpriv.h [hunk 2]
```diff
@@ -344,7 +344,7 @@ class recursive_gc_sync;
 // card marking stealing only makes sense in server GC
 // but it works and is easier to debug for workstation GC
 // so turn it on for server GC, turn on for workstation GC if necessary
-#define FEATURE_CARD_MARKING_STEALING
+//#define FEATURE_CARD_MARKING_STEALING
```
Comment (Maoni0) [inline src/coreclr/gc/gcpriv.h:347]: temporary change?

**Comment by Maoni0** [comment]
> #ifdef SIMPLE_DPRINTF

should we just get rid of SIMPLE_DPRINTF? the only reason I had it was just that stress log didn't take more than 7 params

---
Refers to: src/coreclr/gc/gcpriv.h:264 in ca2a702. [](commit_id = ca2a702d5b777ba7a62d36670cad7846f9416305, deletion_comment = False)

File: src/coreclr/gc/gcpriv.h (hunk 2 above)
Comment (PeterSolMS) [reply src/coreclr/gc/gcpriv.h:347]: Right, did this for regions, thanks for catching it!

**Comment by PeterSolMS** [comment]
> should we just get rid of SIMPLE_DPRINTF? the only reason I had it was just that stress log didn't take more than 7 params
Yes, eventually once we get comfy with the new scheme. Separate PR, I think.

**Comment by AaronRobinsonMSFT** [comment]
Negatively impacted this support in https://github.com/dotnet/runtime/pull/50314. Working with @PeterSolMS offline on a way forward.

**Pull request merged**